### DISCUSSION TITLE
How can I get the address of the wallet connected in WalletConnectUnity ?

### DISCUSSION THREAD

**Discussion by yozdev** [WalletConnect/walletconnect-monorepo#2978] [Developer Support] [answered]
Hello,

While using WalletConnectUnity v1, I was able to access the address of the connected wallet using WCSessionData.accounts[].

How do I access the address of the connected wallet in v2 ?

**Comment by crypblizz8** [comment]
Passed onto our Unity team!

**Reply by gigajuwels** [reply]
The connected addresses can be found in the `SessionData` struct that is given back when waiting for approval 

Example [here](https://github.com/WalletConnect/WalletConnectSharp/blob/2.0/Tests/WalletConnectSharp.Examples/BiDirectional.cs#L91)

**Reply by yozdev** [reply]
Hey @gigajuwels, thanks for replying.

I checked and found the SessionStruct struct. However, I was only able to find the following properties in the struct :

1. Topic
2. Relay
3. Expiry
4. Acknowledged
5. Controller
6. Namespaces
7. RequiredNamespaces
8. Self
9. Peer
10. Key

I also checked the Peer property which is a Participant class type. However, it's only returning PublicKey and Metadata.

How should I get the address of the connected wallet ?

**Reply by gigajuwels** [reply, marked as answer]
It would be under `SessionStruct.Namespaces[chainID].Accounts`

`Namespaces` is a dictionary field of all the connected blockchain networks and accounts in the given session.

 ```
    /// <summary>
    /// A dictionary of Namespaces based on a chainId key. The chainId
    /// should follow CAIP-2 format
    /// chain_id:    namespace + ":" + reference
    /// namespace:   [-a-z0-9]{3,8}
    /// reference:   [-_a-zA-Z0-9]{1,32}
    /// </summary>
```

**Reply by yozdev** [reply]
Thank you @gigajuwels 

One thing I would like to mention.

I observed that we have to pass the chainID evm namespace instead of chainID. Passing complete chainID is returning the following error

System.Collections.Generic.KeyNotFoundException: The given key 'eip155:137' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <4a4789deb75f446a81a24a1a00bdd3f9>:0 

However, when I use SessionStruct.Namespaces["eip155"].Accounts, I get the address and no errors.
